gameobjects/scorearea.cpp

```diff
@@ -11,24 +11,31 @@
 namespace pong
 {
 
-ScoreArea::ScoreArea(float width, float height, bool playerSide) :
+ScoreAreaBlueprint::ScoreAreaBlueprint(float width, float height, bool playerSide) :
     mWidth { width },
     mHeight { height },
     mIsPlayerScoreArea { playerSide }
 {
     SetInstanceName("ScoreArea");
+    InitalizeComponents();
```

Comment:
So I did opt to just call `InitalizeComponents()` from within the constructor. However I am curious if is it even worth doing it this way instead of calls to `AddComponent<>()` in the constructor itself. With this current method it forces `InitalizeComponents` to be called and requires saving the parmeters from the constructor for the creation of the components later on.

Reply:
Good question. I think I would likely just flatten it together in the Constructor. I see what you're trying to do by forcing all your subclasses to initialize, but I think it is just boilerplate that doesn't actually prevent a bug.

I'll ponder if there is any way to enforce what you want to enforce, nothing is coming to mind right now...